app/controllers/users/registrations_controller.rb

```diff
@@ -2,4 +2,5 @@
 
 class Users::RegistrationsController < Devise::RegistrationsController
   include Devise::Passkeys::Controllers::RegistrationsControllerConcern
+  include RelyingParty
```

Comment:
@tcannonfodder : I've noticed that in the `devise-passkeys` gem README.md, this controller have a method called `relying_party` with `WebAuthn::RelyingParty.new(...)`.

Looking at the template app, it looks to me that this controller concern (app > controllers > concern > relying_party.rb) could have a default file in the `devise-passkeys` gem itself, read some attributes from `ENV` (at least origin and name), so as a user I don't need to have create one on each app, just to change the basic values for origin and name.

Later on, this setup could even an initializer, in the app itself, so this concern (RelyingParty) would really be unchanged and not replicated in every app.

Do you value in that approach? Or do you foresee any issues with it?

BTW: I don't think this should be the scope of this PR, but more a long term strategy.

Reply:
I just realized one thing: for prod/staging environments, RP origin will be different, so that definitively needs from come from the environment, correct?

Anyway, if that is the way to go for production, we can work on it as another PR.

Reply:
On my phone, apologies in advance 📲

So, the reasons I haven't included automatic relying party generation in the library are
- The principle of least magic
- Ruby-WebAuthn ships with good defaults & I don't want to muck with that
- Each app is different
- The Relying Party is the most crucial part of the authentication chain, so the implementation should be in full control over it

Basically, every app has its own flavor of authentication over time. Some examples: subdomained apps, apps that use different warden strategies for different user types, a microservice or middleware that handles just authentication.

Not only does Ruby WebAuthn provide dynamic Relying Party setup (which is great because it allows for handling all of these cases!), it also sets up relying parties with smart defaults out of the gate.

Given how crucial authentication is for apps, and how important the Relying Party is to the WebAuthn chain, I think we need to adhere to the principle of least magic here. Especially since the underlying library has smart defaults. Having an undefined `relying_party` method explicitly states: "hey, we've done as much setup as we feel assured of doing. We can't make assumptions about your app, please provide us with the core configuration"

Essentially: devise-passkeys focuses its efforts on making registration & authentication streamlined and as strong as possible. In the future it will hopefully have some guidance & helpers for secondary problems like recovery processes. But to avoid misconfigurations or frustration, we leave the relying party setup up to the implementor